mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergSplit.java

```diff
@@ -73,7 +74,10 @@ public String[] getLocations() {
     // getLocations() won't be accurate when called on worker nodes and will always return "*"
     if (locations == null && conf != null) {
       boolean localityPreferred = conf.getBoolean(InputFormatConfig.LOCALITY, false);
-      locations = localityPreferred ? Util.blockLocations(task, conf) : ANYWHERE.clone();
+      locations =
+          localityPreferred
+              ? Util.blockLocations((ScanTaskGroup<FileScanTask>) task, conf)
```

Comment:
I believe this change isn't necessary and it should also work without the cast

Reply:
You are right. I guess IDE added it automatically. Removed redundant cast.